src/lightning_lite/plugins/collectives/torch_collective.py

```diff
@@ -0,0 +1,169 @@
+    ) -> Optional[List[torch.Tensor]]:
+        dist.gather(tensor, gather_list, dst, group=self.group)
+        return gather_list
+
+    def scatter(
+        self, tensor: torch.Tensor, scatter_list: Optional[List[torch.Tensor]] = None, src: int = 0
+    ) -> torch.Tensor:
+        dist.scatter(tensor, scatter_list, src, group=self.group)
+        return tensor
+
+    def reduce_scatter(
+        self, output: torch.Tensor, input_list: List[torch.Tensor], op: Union[str, ReduceOp] = "sum"
+    ) -> torch.Tensor:
+        op = self._convert_to_native_op(op)
+        dist.reduce_scatter(output, input_list, op=op, group=self.group)
+        return output
+
+    def all_to_all(
+        self, output_tensor_list: List[torch.Tensor], input_tensor_list: List[torch.Tensor]
+    ) -> List[torch.Tensor]:
+        dist.all_to_all(output_tensor_list, input_tensor_list, group=self.group)
+        return output_tensor_list
+
+    def send(self, tensor: torch.Tensor, dst: int, tag: Optional[int] = 0) -> None:
+        dist.send(tensor, dst, tag=tag, group=self.group)
+
+    def recv(self, tensor: torch.Tensor, src: Optional[int] = None, tag: Optional[int] = 0) -> torch.Tensor:
+        dist.recv(tensor, src, tag=tag, group=self.group)
+        return tensor
+
+    def all_gather_object(self, object_list: List[Any], obj: Any) -> List[Any]:
+        dist.all_gather_object(object_list, obj, group=self.group)
+        return object_list
+
+    def broadcast_object_list(
+        self, object_list: List[Any], src: int, device: Optional[torch.device] = None
+    ) -> List[Any]:
+        kwargs = {}
+        if _TORCH_GREATER_EQUAL_1_10:
+            kwargs["device"] = device
+        dist.broadcast_object_list(object_list, src, group=self.group, **kwargs)
+        return object_list
+
+    def gather_object(
+        self, obj: Any, object_gather_list: Optional[List[Any]] = None, dst: int = 0
+    ) -> Optional[List[Any]]:
+        dist.gather_object(obj, object_gather_list, dst, group=self.group)
+        return object_gather_list
+
+    def scatter_object_list(
+        self, scatter_object_output_list: List[Any], scatter_object_input_list: Optional[List[Any]], src: int = 0
+    ) -> List[Any]:
+        dist.scatter_object_list(scatter_object_output_list, scatter_object_input_list, src, group=self.group)
+        return scatter_object_output_list
+
+    def barrier(self, device_ids: Optional[List[int]] = None) -> None:
+        dist.barrier(group=self.group, device_ids=device_ids)
+
+    def monitored_barrier(self, timeout: Optional[datetime.timedelta] = None, wait_all_ranks: bool = False) -> None:
+        dist.monitored_barrier(group=self.group, timeout=timeout, wait_all_ranks=wait_all_ranks)
+
+    def setup(
+        self, main_address: Optional[str] = None, main_port: Optional[str] = None, **kwargs: Any
+    ) -> Self:  # type: ignore[valid-type]
+        if not self.is_initialized():
```

Comment:
why do we need to set those? Once we initialized the group they should not matter anymore